File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0; // keep track of unique
        for(int j=1; j<nums.size(); j++) {
            if(nums[i] != nums[j]) {
                i++;
                nums[i] = nums[j];
            }
        }
        return i+1; // this equals the no of unique elements
    }
};

/*      
set<int> s;
for(int i=0; i<nums.size(); i++) {
    s.insert(nums[i]);
}

int k = s.size();
int i = 0;
for(auto n: s) {
    nums[i] = n;
    i++;
}

return k; 
*/

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int k = 0;
        int currElem = 0;
        int freq = 0;
        for(int i=0; i<nums.length; i++) {
            if(nums[i] == currElem) {
                freq++;
                if(freq <= 1) k++;
            }
            else {
                nums[k] = nums[i];
                currElem = nums[i];
                freq = 1;
                k++;
            }
        }
        return k;
    }
}